modules/installation/pages/upgrade.adoc

```diff
@@ -92,42 +92,74 @@ $ gadmin restart all -y
 
-[NOTE]
-====
-Rolling Upgrade is only available for HA clusters (High Availability) and when making a maintenance update (e.g., 4.2.0 to 4.2.1, 4.3.1 to 4.3.3). 
-Upgrading from 4.2.0 to 4.3.0 or any other major version is not supported.
-====
+*Before You Begin*
+
+Rolling Upgrade is only available for HA (High Availability) clusters.
+It is supported only for maintenance updates (for example, 4.2.0 to 4.2.1, or 4.3.1 to 4.3.3). Upgrading from 4.2.0 to 4.3.0 or another major version is not supported.
+
+During a rolling upgrade:
+
+* Loading jobs and running queries will continue to function.
+* Occasional slowdowns in queries per second (QPS) may occur.
+* TigerGraph does not upgrade Zookeeper (ZK), Kafka, or ETCD as part of this upgrade.
+
+*Query Processing and Failure Handling During Rolling Upgrade*
+
+Queries usually continue to work while a rolling upgrade is in progress, but can fail in these scenarios:
+
+* If a query is sent to a replica that is about to be upgraded, it may fail because the services on that replica will be restarted. The query will timeout if the replica doesn’t respond in time.
+
+* Queries may fail if their execution time exceeds the GPE stop timeout.
+By default, `GPE.StopTimeoutMS` is set to 5 minutes; after this time, GPE is forcefully stopped and any incomplete queries will fail.
+
+TigerGraph handles queries during the upgrade in the following ways:
+
+* Upon receiving a query, RESTPP checks that every partition has at least one live replica available. If any partition lacks a live replica, the query fails immediately.
+
+* RESTPP forwards each query to the set of live replicas in a round-robin fashion (for example, in a 2x3 cluster, if two nodes are being upgraded, queries will go to the remaining four nodes).
+
+* The query succeeds only if all replicas that received the query respond successfully. If any of them restarts before responding, the query fails.
+
+*Best Practices*
+
+To minimize query failures and keep your database highly available during a rolling upgrade, follow these recommendations:
+
+* Implement automatic retries with round-robin logic and backoff for failed queries.
+* Avoid running long queries during upgrade. Break them into smaller parts or optimize them when possible.
+* Adjust `GPE.StopTimeoutMS` if your use case requires longer-running queries (be aware that this will increase overall upgrade time, as GPE waits longer before stopping).
 
 *Blocked Operations During Rolling Upgrade*
 
 To ensure a smooth upgrade, the following operations are blocked during a rolling upgrade:
 
-* Install query
-* Run schema change
-* Drop all
-* Run `gsql --reset`
-* Create graph
-* Clear graph store
-* Export or import graph
-* Install function
+* `INSTALL QUERY`
+* `SCHEMA CHANGE`
+* `DROP ALL`
+* `gsql --reset`
+* `CREATE GRAPH`
+* `CLEAR GRAPH STORE`
+* `EXPORT` or `IMPORT GRAPH`
+* `INSTALL FUNCTION`
```

Comment:
The original list was written as English descriptions of features; that is why it was in lowercase.
If we capitalize the items, then we are writing code, and the code has to be correct. Now if I look at it from a code perspectively, I see a lot of contextual inconsistencies. That is, the developer created a problem by holding us to a higher standard without fixing the code.